cpp/src/strings/convert/is_valid_element.cu

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2021, Baidu CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/strings/detail/utilities.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <strings/utilities.cuh>
+#include <strings/utilities.hpp>
+#include <strings/utilities.cuh>
+
+#include <thrust/logical.h>
+
+namespace cudf {
+namespace strings {
+
+/**
+ * Check whether the UTF8String is valid when convert data from string to all kinds of integers,
+ * like INT8/16/32/64. for example, if allow_decimal is true, this will return `true, true` when input string 
+ * is `1.23, 123`, or this function will return `false, true`, it means firt element is invalid, 
+ * while second data is valid. 
+ *
+ * Note that, in this method we accumulate the result in negative format, and convert it to
+ * positive format at the end, if this string is not started with '-'. This is because min value
+ * is bigger than max value in digits, e.g. Long.MAX_VALUE is '9223372036854775807' and
+ * Long.MIN_VALUE is '-9223372036854775808'.
+ *
+ * This code is mostly copied from LazyLong.parseLong in Hive.
```

Comment:
Wouldn't we need to get permission to copy code from another source into our repo?
What is the license agreement at the source? We may need a SWIPAT?

Reply:
The description here is not very accurate , source code is java , while here is c++ , so I don't know if I need permission.
this code is motly copied from https://github.com/apache/spark/blob/cc1d9d25fb4c2e4af912d6f9802de8f351c32deb/common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java#L1116-L1194
and from notes i see  that code is mostly copied from LazyLong.parseLong in Hive.

Reply:
The source code language would only matter for copyright I think. 
The license at the top of the source file should identify how the code may be used.
@fondaing @harrism @kkraus14

Reply:
The license in both cases is the same Apache License. I am not a lawyer, so if you need to check with someone about this you can. I believe that taking code from one Apache Licensed project to use in another should not be an issue. You can take Apache Licensed code and use it in a proprietary application without releasing the source code. Giving credit for where the code came from is not required from the Apache license, but it is a nice thing to do. I personally think it would be more accurate to say that the code here is heavily based off of LazyLong.parseLong from Hive, but updated for C++.